flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -263,7 +269,8 @@ BufferAndBacklog pollBuffer() {
 
 				buffer = bufferConsumer.build();
 
-				checkState(bufferConsumer.isFinished() || buffers.size() == 1,
+				checkState(bufferConsumer.isFinished() || buffers.size() == 1 ||
```

Comment:
Why do we need that change here and not for aligned barriers?

Reply:
Reverted this change - will fix FLINK-19907 instead.

Reply:
And thanks for this question! :+1: